### PR TITLE
[DOCS] Adds documentation for ACF field naming convention

### PR DESCRIPTION
**Pull request by andrewjt71**

### CODE REVIEW

File: docs/dev/custom-code.md [hunk 1]
```diff
@@ -126,3 +126,32 @@ Under the hood this runs the following commands:
...
+
+### Example
+
+If you have a block called accordion, which has a "drawer" repeater field, which consists of multiple accordion items, each of which has a title field, the field list would be:
+- Accordion drawer: `accordion_block__drawer`
+- Accordion item summary: `summary`
+
+Retrieving the values looks like this: 
+
+```
```
Comment (jenkoian) [inline docs/dev/custom-code.md:149]: Can you do...

```
```twig
...
```
```

??

File: docs/dev/custom-code.md [hunk 2]
```diff
@@ -126,3 +126,32 @@ Under the hood this runs the following commands:
...
+{% set drawer = [] %}
+
+{% for item in fields['accordion_block__drawer'] %}
+    {% set drawer = drawer | merge([{
+        summary: item['summary']
+    }]) %}
+{% endfor %}
+```
```
Comment (jenkoian) [inline docs/dev/custom-code.md:157]: Worth having a php example too?

File: docs/dev/custom-code.md (hunk 1 above)
Comment (andrewjt71) [reply docs/dev/custom-code.md:149]: I can indeed!

File: docs/dev/custom-code.md (hunk 2 above)
Comment (andrewjt71) [reply docs/dev/custom-code.md:159]: Done!

File: docs/dev/custom-code.md (hunk 1 above)
Comment (andrewjt71) [reply docs/dev/custom-code.md]: Done!

File: docs/dev/custom-code.md [hunk 3]
```diff
@@ -126,3 +126,43 @@ Under the hood this runs the following commands:
...
+
+PHP:
+```php
```
Comment (jenkoian) [inline docs/dev/custom-code.md:162]: Extremely minor but newline after `PHP:` as per the twig one above.

File: docs/dev/custom-code.md [hunk 4]
```diff
@@ -126,3 +126,43 @@ Under the hood this runs the following commands:
...
+$drawer = [];
+
+foreach (get_field('accordion_block__drawer') as $item) {
+  $drawer[] = [ 'summary' => $item['summary']];
+}
+```
```
Comment (jenkoian) [inline docs/dev/custom-code.md:168]: Can we add spaces as per the WP standards, e.g. `foreach ( get_field(...`

File: docs/dev/custom-code.md [hunk 5]
```diff
@@ -126,3 +126,43 @@ Under the hood this runs the following commands:
...
+    }]) %}
+{% endfor %}
+```
+
+PHP:
+```php
+$drawer = [];
+
+foreach (get_field('accordion_block__drawer') as $item) {
+  $drawer[] = [ 'summary' => $item['summary']];
```
Comment (jenkoian) [inline docs/dev/custom-code.md:166]: Space before the closing `]` at the end of this.

File: docs/dev/custom-code.md (hunk 3 above)
Comment (andrewjt71) [reply docs/dev/custom-code.md:163]: Done, ta

File: docs/dev/custom-code.md (hunk 4 above)
Comment (andrewjt71) [reply docs/dev/custom-code.md:169]: Yep makes sense, done!

File: docs/dev/custom-code.md (hunk 5 above)
Comment (andrewjt71) [reply docs/dev/custom-code.md]: Oops, done!

**Pull request merged**